content/en/docs/concepts/architecture/nodes.md

```diff
@@ -98,6 +99,14 @@ When the [Node authorization mode](/docs/reference/access-authn-authz/node/) and
 [NodeRestriction admission plugin](/docs/reference/access-authn-authz/admission-controllers/#noderestriction) are enabled,
 kubelets are only authorized to create/modify their own Node resource.
 
+{{< note >}}
+As mentioned in the [Node name uniqueness](#node-name-uniqueness) section,
+when Node configuration needs to be updated, it is a good practice to re-register
+the node with the API server. For example, if the kubelet being restarted with
+the new set of `--node-labels`, but the same Node name is used, the change will
+not take an effect, as labels are being set on the Node registration.
```

Comment:
Is there a reason for this limitation (eg information security)? I'm wondering how come a running kubelet isn't able to relabel the Node that represents it.

Reply:
I don't know the history. I think it is to avoid situation when pods get incorrectly-scheduled due to taints on the kubelet restart. Draining and re-adding node works around this issue

Reply:
do you think adding the reasoning in the text will be helpful for readers?